### PR TITLE
PLU-115: [SST-2] Fetch test execution steps with new query

### PR DESCRIPTION
**Pull request by pregnantboy**
### TL;DR

Replaces the repeated lazy querying of execution test results with the new query 'GetTestExecutionSteps`. This also ensures the test variables are still present even after user exits the editor.

### What changed?

1. Removed repeated calls of the old query whenever an accordion is expanded
2. Calls the new `GetTestExecutionSteps` query in `EditorContext` for managing test step execution states.
3. Updated `FlowStep`, `TestResult`, `TestSubstep`, and other affected components to use `testExecutionSteps`.

### How to test?

1. Run the app
2. Navigate to a pre-tested pipe. 
3. Expand accordions to see test results / error details.

Note: refreshing of data upon re-test steps will be introduced in the next PR

### CODE REVIEW

**Comment by pregnantboy** [comment]
* **#724** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/724?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#708** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/708?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#707** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/707?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#706** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/706?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#680** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/680?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#679** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/679?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#678** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/678?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `develop-v2`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @pregnantboy and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/frontend/src/components/FlowStep/index.tsx [hunk 1]
```diff
@@ -102,32 +101,17 @@ export default function FlowStep(
...
-      // the user toggles a step. Let's fix this in a separate PR.
-      getStepWithTestExecutions({
-        variables: {
-          stepId: step.id,
-        },
-      })
-    }
-  }, [collapsed, getStepWithTestExecutions, step.id, isTrigger])
 
+  // This includes all steps that run even after the current step, but within the same branch.
```
Comment (m0nggh) [inline packages/frontend/src/components/FlowStep/index.tsx:106]: nice thks for the comment

File: packages/frontend/src/helpers/variables.ts [hunk 2]
```diff
@@ -113,26 +113,28 @@ const process = (data: any, parentKey?: any, index?: number): RawVariable[] => {
...
-      return hasExecutionSteps
+  return executionSteps
+    .filter((executionStep: IExecutionStep) => {
+      const hasDataOut = Object.keys(executionStep.dataOut ?? {}).length
+      return hasDataOut
     })
-    .map((step: IStep, index: number) => ({
-      id: step.id,
+    .map((executionStep: IExecutionStep, index: number) => ({
+      id: executionStep.id,
```
Comment (m0nggh) [inline packages/frontend/src/helpers/variables.ts]: can i check what's the reason for changing from `step.id` to the `executionStep.id` ah

File: packages/frontend/src/components/TestSubstep/index.tsx [hunk 3]
```diff
@@ -61,31 +61,31 @@ function TestSubstep(props: TestSubstepProps): JSX.Element {
...
       (variable) => (variable.type ?? 'text') === 'text',
     )
-  }, [executionStep])
-
-  const isExecuted = !error && called && !loading
-  const isCompleted = isExecuted && response
+    if (stepWithVariables.length > 0) {
+      return stepWithVariables[0].output
+    }
+    return []
```
Comment (m0nggh) [inline packages/frontend/src/components/TestSubstep/index.tsx:93]: should this be null?

**Review by m0nggh — APPROVED** [review]
lgtm thks for the refactor, remember to pull the latest changes for `variables` before merging :-)

**Comment by pregnantboy** [comment]
### Merge activity

* **Aug 27, 6:30 AM EDT**: @pregnantboy started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/opengovsg/plumber/679).
* **Aug 27, 6:34 AM EDT**: [Graphite](https://app.graphite.dev/github/pr/opengovsg/plumber/679) rebased this pull request as part of a merge.
* **Aug 27, 6:35 AM EDT**: @pregnantboy merged this pull request with [Graphite](https://app.graphite.dev/github/pr/opengovsg/plumber/679).
